### DISCUSSION TITLE
`.content-visible` div dependent on `params`?

### DISCUSSION THREAD

**Discussion by r-cheologist** [quarto-dev/quarto-cli#3116] [Q&A] [answered]
I'd like to do something like:

    ---
    params:
      show_private_info: true
    ---
    
    ::: {.content-visible <ARGUMENT TESTING FOR `params$show_private_info`>}
    Some private information
    :::

Is that feasible?

**Comment by r-cheologist** [comment]
Moving this to [stackexcange.com](https://stackoverflow.com/q/74261795/2103880). Don't seem to be able to delete here. Sorry about that.

**Comment by nclJoshCowley** [comment, marked as answer]
I currently use the following to hide out whole sections in my reports.

```r
`r if (params$show_private_info) "::: {.content-hidden}"`

Some private information

`r if (params$show_private_info) ":::"`
```

But [project profiles](https://quarto.org/docs/projects/profiles.html) are probably a better solution I've yet to try. They give this example in the docs.

```
::: {.content-visible when-profile="advanced"}
This content will only appear in the advanced version.
:::
```

**Reply by r-cheologist** [reply]
Thank you for your input, which works indeed.

Your point regarding `profile`s is well taken and I'm looking at how to use profile default definition to gain as clean a fine grained interface in `_quarto.yml` as with the `parameters` definition ...
